ovve_ui/utils/comms_link.py

```diff
@@ -0,0 +1,278 @@
+import json
+import random
+import time
+from threading import Thread
+import serial
+import sys
+from time import sleep
+import binascii
+import crc16
+import codecs
+
+from utils.comms_adapter import CommsAdapter
+from utils.params import Params
+from utils.settings import Settings
+
+
+class CommsLink():
+    def __init__(self, comms_adapter: CommsAdapter) -> None:
+        self.comms_adapter = comms_adapter
+        self.comms_adapter.set_comms_callback(self.update_settings)
+        self.done = False
+        self.settings = Settings()
+        self.seqnum = 0
+        self.packet_version = 1
+        self.BAUD = 38400
+        self.PORT = "/dev/ttyACM0"
```

Comment:
Is the serial port always the same?   I think it would be better to encapsulate serial settings somewhere else.

Reply:
I am hoping that this doesn't change so much. Can I change this later when the mechanism for getting those settings are decided. It's basically an operation that deals with setup

Reply:
If there is an initial setup process. I can get the settings from there